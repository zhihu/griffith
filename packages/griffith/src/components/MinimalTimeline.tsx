import React, {Component} from 'react'
import PropTypes from 'prop-types'

import Slider from './Slider'

import {slider, minimal} from './Timeline.styles'

class MinimalTimeline extends Component {
  static propTypes = {
    duration: PropTypes.number.isRequired,
    currentTime: PropTypes.number.isRequired,
    buffered: PropTypes.number.isRequired,
    show: PropTypes.bool,
    progressDots: PropTypes.arrayOf(
      PropTypes.shape({
        startTime: PropTypes.number.isRequired,
      })
    ),
  }

  static defaultProps = {
    duration: 0,
    currentTime: 0,
    buffered: 0,
    progressDots: [],
  }

  shouldComponentUpdate(nextProps) {
    return this.props.show || nextProps.show
  }

  render() {
    const {duration, currentTime, buffered, progressDots} = this.props
    return (
      <Slider
        value={currentTime}
        total={duration}
        buffered={buffered}
        progressDots={progressDots}
        styles={[slider, minimal]}
        noInteraction
      />
    )
  }
}

export default MinimalTimeline
