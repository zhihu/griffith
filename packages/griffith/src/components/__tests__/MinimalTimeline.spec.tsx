import React from 'react'
import {render} from '@testing-library/react'
import MinimalTimeline from '../MinimalTimeline'

describe('MinimalTimeline', () => {
  it('get MinimalTimeline component', () => {
    const result = render(
      <MinimalTimeline
        buffered={0}
        duration={182.234}
        currentTime={0}
        show={false}
      />
    )

    expect(result.container).toMatchSnapshot()

    result.rerender(
      <MinimalTimeline
        buffered={0}
        duration={182.234}
        currentTime={60}
        show={true}
      />
    )

    expect(result.container).toMatchSnapshot()
  })
})
