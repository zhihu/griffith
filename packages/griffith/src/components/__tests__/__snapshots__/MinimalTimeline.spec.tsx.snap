// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalTimeline get MinimalTimeline component 1`] = `
<div>
  <div
    class="root_17bkmwy-o_O-root_1u52a97-o_O-root_1pr59fm-o_O-root_qckecg"
    role="slider"
  >
    <div class="inner_9gm5i8-o_O-inner_qb00m9">
      <div
        class="
          track_1bopucd-o_O-track_2wh41e-o_O-track_1q3h7g7-o_O-track_1h7ofbx
        "
      >
        <div
          class="bar_1kpj1ed-o_O-bar_g2nzkb-o_O-bar_1kvrquw"
          style="left: 0px; width: 0%"
        ></div>
      </div>
    </div>
  </div>
</div>

`;

exports[`MinimalTimeline get MinimalTimeline component 2`] = `
<div>
  <div
    class="root_17bkmwy-o_O-root_1u52a97-o_O-root_1pr59fm-o_O-root_qckecg"
    role="slider"
  >
    <div class="inner_9gm5i8-o_O-inner_qb00m9">
      <div
        class="
          track_1bopucd-o_O-track_2wh41e-o_O-track_1q3h7g7-o_O-track_1h7ofbx
        "
      >
        <div
          class="bar_1kpj1ed-o_O-bar_g2nzkb-o_O-bar_1kvrquw"
          style="left: 0px; width: 32.92470120833653%"
        ></div>
      </div>
    </div>
  </div>
</div>

`;
